### DISCUSSION TITLE
Columns Organized by npub Clusters

### DISCUSSION THREAD

**Discussion by tkhumush** [damus-io/notedeck#295] [Ideas]
GM Damus Team. 😄 

## Something New (Maybe)
I was thinking about the two views in notedeck, the column view and the avatar bubbles view. And wondering how those could be combine to provide a new social experience. Maybe you are already thinking of this. If anything maybe I can contribute with the UX idea.

## Under The Hood
Using [Organizational Network Analysis](https://github.com/wangpengcn/Organizational-network-analysis) (ONA) clusters of conversations could be identified based on certain criteria and thresholds. - The result of the Analysis could be an event by itself and the user could decide if they wants to share it.

## The User Experience
The user gets a notification **"Your friends (x,y,z) are talking... Add Column?"**. This could either be generated by the users own notedeck + nostrdb or shared as a conversation event by an npub the user follows. I don't think there is any website that shares a discussion - maybe reddit? It might be similar to reddit, but faster, flexible, and unbounded by the limits of a Reddit Post.

These clusters happen all the time on nostr. Adding discoverability to these conversations by creating an event that could be shared might be interesting and fun. (And of course way more difficult that i imagine) 🤣 

<img width="935" alt="image" src="https://github.com/user-attachments/assets/d5109d74-9303-44f7-baf4-d8ee2e64e6b2">

## Maybe there is an easier approach?
I think there is room for creating a quick and dirty version of this idea by focusing on creating columns for threads that get a certain number of "re-shares" and "quotes". This is much simpler and might achieve similar results. The key difference here, the focus is not the note event it self, but who are in the discussion. and the column is created around the topic even if it starts to diverge from the original note.

**Comment by alltheseas** [comment]
Nice! Lists are a simple and static start to having groups of folks.

You are suggesting surfacing a discussion from a certain group of people. 

If this were an algorithm, how would you define it?

**Reply by tkhumush** [reply]
Yes, Lists pull npubs regardless of what the notes are about.  

This clusters conversations. Highlights conversations happening now. 

Also, I think this is not just an algorithm. An algorithm works in one dimension, mostly to surface specific Notes with certain engagement. 

These conversation clusters is like entering a Bar and scanning it to see where your friends are at then go join them.

**Reply by jb55** [reply]
We just need to assign a score (aka distance metric) between any two npubs. This can be calculated in many different ways. The most obvious way is people who follow who, but could include people who interact with each other a lot.

Ideally this metric is configurable and updatable, pluggable, so that you can just create generic visualizations based on different metrics.

**Comment by jb55** [comment]
This is very close to what I was thinking! I wanted a visualization view where you can basically circle a group of people and create a list from that.
